### ISSUE TITLE
MTL with python

### ISSUE THREAD

**Issue by sripar** [Microsoft/CNTK#1616]
I get a Runtime error while trying to combine two different losses 

A Function instance with more than one output cannot be implicitly converted to a Variable

What is the efficient way of training a network with multiple outputs or multiple tasks ?

**Comment by bencherian** [comment]
Create a function that appropriately weights the multiple losses that takes your multiple losses as inputs. It could just be as simple as adding the losses together. (If the loss function has some sort of scale to it, you likely want to weight the losses accordingly so one of the outputs or tasks doesn't overwhelm the others.)

**Comment by sripar** [comment]
I was wondering if there was an efficient way to combine models for training with python
The combine method seems to be causing the Runtime error

**Comment by bencherian** [comment]
That doesn't entirely make sense...if you have multiple loss functions connected to the same parameters you have to indicate how to handle tradeoffs when optimizing both of them. (For example, suppose the gradient w.r.t. loss1 tells you to decrease a parameter, while the gradient w.r.t. loss2 tells you to increase a parameter, what would you do?) The simplest solution is just to add them all together and optimize that function. If they are connected to completely disparate sets of parameters, they might as well be completely separate models.

**Comment by sripar** [comment]
My problem is not the formulation itself. I am able to do this perfectly with brainscript i.e weighing different losses for MTL etc. My problem is to do this in python. A specific way to combine multiple models (combine function) as shown in the link doesn't seem to work . https://github.com/Microsoft/CNTK/wiki/Train-two-or-more-models-jointly

**Comment by bencherian** [comment]
Are you combining the losses or the models? You should add the losses and combine the actual networks that are producing estimated outputs.

**Comment by sripar** [comment]
That's exactly what I am doing. Adding the losses, combining network

**Comment by sayanpa** [comment]
Closing this issue since there was a suggestion and no further communication on this thread. Please reopen the issue with details if you need a follow up.
